pandas/core/internals/blocks.py

```diff
@@ -1971,6 +1982,12 @@ def convert(
         if self.dtype != _dtype_obj:
             # GH#50067 this should be impossible in ObjectBlock, but until
             #  that is fixed, we short-circuit here.
+            if using_copy_on_write:
+                result = self.copy(deep=False)
+                result._ref = weakref.ref(  # type: ignore[attr-defined]
+                    original_blocks[self.mgr_locs.as_array[0]]
```

Comment:
I was going to say: "isn't `original_blocks[..]` just `self`?". But that's not the case because of `maybe_split`?